scripts/convert_yaml_to_sdf.py

```diff
@@ -117,10 +121,46 @@ def read_geometry(shape_item):
         if "pose" in yml_box:
             link_pose = read_pose(yml_box)
 
+    if "path" in shape_item and "blockheight" in shape_item:
+
+        # If there is a path and a blockheight in the shape_item, then there is a heightmap included in the yaml file
+        image_path = os.getenv("ED_MODEL_PATH") + "/{}".format(model_name) + "/{}".format(shape_item["path"])
+        new_image_path = os.path.dirname(image_path) + "/{}_converted.png"\
+            .format(os.path.splitext(shape_item["path"])[0])
+
+        # Execute Imagemagick command to invert the image
+        call("convert -negate {} {}".format(image_path, new_image_path), shell=True)
+
+        # Import the new png image and get its sizes to determine the physical square length (in meters) of the map
+        with Image.open(new_image_path, "r") as f:
+            width, height = f.size
+        resolution = int(pow(2, (max(width, height)-2).bit_length())+1)
```

Comment:
The -2 is needed so that a e.g. a 4097x2000 image won't be scaled to 8193x8193 but to 4097x4097, but that 4098x2000 will be scaled to 8193. So it just makes sure that border-cases don't get unnecessarily upscaled when they would already fit in a Gazebo standard raster.

Reply:
Maybe good to add as comment